Source/WebKit/UIProcess/Inspector/mac/RemoteWebInspectorUIProxyMac.mm

```diff
@@ -49,17 +49,22 @@
 #import <wtf/text/Base64.h>
 
 @interface WKRemoteWebInspectorUIProxyObjCAdapter : NSObject <NSWindowDelegate, WKInspectorViewControllerDelegate> {
-    WebKit::RemoteWebInspectorUIProxy* _inspectorProxy;
+    WeakPtr<WebKit::RemoteWebInspectorUIProxy> _inspectorProxy;
 }
 - (instancetype)initWithRemoteWebInspectorUIProxy:(WebKit::RemoteWebInspectorUIProxy*)inspectorProxy;
 @end
 
 @implementation WKRemoteWebInspectorUIProxyObjCAdapter
 
+- (RefPtr<WebKit::RemoteWebInspectorUIProxy>)protectedInspectorProxy
+{
+    return _inspectorProxy.get();
+}
+
 - (NSRect)window:(NSWindow *)window willPositionSheet:(NSWindow *)sheet usingRect:(NSRect)rect
 {
-    if (_inspectorProxy)
-        return NSMakeRect(0, _inspectorProxy->sheetRect().height(), _inspectorProxy->sheetRect().width(), 0);
+    if (RefPtr inspectorProxy = _inspectorProxy.get())
+        return NSMakeRect(0, inspectorProxy->sheetRect().height(), inspectorProxy->sheetRect().width(), 0);
```

Comment:
`RemoteWebInspectorUIProxy::sheetRect` is a simple getter (i.e. it returns a member variable).  is it still necessary to protect it?  i thought we only were concerned about non-simple logic?

Reply:
Oh I guess we don't need it in that case.